closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -1661,30 +1763,55 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_RSADP
     msg_len = (*OSSL_RSA_private_encrypt)(kLen, kNative, mNative, rsaKey, RSA_NO_PADDING);
 
     if ((-1 != verify) && (-1 != msg_len)) {
-        if (verify == kLen) {
-            k2 = malloc(kLen * (sizeof(unsigned char)));
-            if (NULL != k2) {
-
-                /* mNative is size 'verify' */
-                msg_len2 = (*OSSL_RSA_public_decrypt)(verify, mNative, k2, rsaKey, RSA_NO_PADDING);
-                if (-1 != msg_len2) {
-
-                    int i;
-                    for (i = 0; i < verify; i++) {
-                        if (kNative[i] != k2[i]) {
-                            msg_len = -2;
-                            break;
+        if (verify == msg_len) {
+            if(-1 != rsadpBNptr) {
+                rsadp_bn_ptr = (RSADP_BN *)rsadpBNptr;
+                /* We are using OpenSSL 1.1.*
+                 * Verify by comparing BigNum of mNative and m_bigNum ^ e % n */
+                const BIGNUM *n = NULL;
+                const BIGNUM *e = NULL;
+                (*OSSL_RSA_get0_key)(rsaKey, &n, &e, NULL);
+                (*OSSL_BN_bin2bn)(kNative, kLen, rsadp_bn_ptr->k_bigNum);
+                (*OSSL_BN_bin2bn)(mNative, msg_len, rsadp_bn_ptr->m_bigNum);
+                if (NULL != rsadp_bn_ptr->k_bigNum && NULL != rsadp_bn_ptr->m_bigNum && NULL != e && NULL != n) {
```

Comment:
Four sets of missing parentheses.

Reply:
You meant the ones in the if statement right? So it should be like this: `if ((NULL != rsadp_bn_ptr->k_bigNum) && (NULL != rsadp_bn_ptr->m_bigNum) && (NULL != e) && (NULL != n)) {`

Reply:
Yes.

Reply:
Okay, then this is fixed now.

Reply:
Okay, I changed that.